src/acl/agent.ts

```diff
@@ -386,6 +386,21 @@ function removeAgentFromResourceRule(
     acl.accessTo,
     resourceIri
   );
+  const agentHasDefaultAccess =
+    getIriAll(rule, acl.agent).filter(
+      (originalAgent) => originalAgent === agent
+    ).length > 0;
+  if (!agentHasDefaultAccess) {
+    // If the original rule if both a resource and a default rule, and did **not**
+    // give default access to the agent whose access are being set, only the resource
+    // access should be duplicated (not the default ones)
+    ruleForOtherTargets = removeIri(
+      ruleForOtherTargets,
+      acl.default,
+      resourceIri
+    );
```

Comment:
Wouldn't this mean that, if the rule for other targets has an `acl:default` rule _and_ applies to other agents/agent classes/agent groups, that those other agents would no longer have default access any more either?

Reply:
Ah darn you're right. Then I guess the simpler approach is to split the rules that are both resource and default into two separate rules, and then apply the original logic to the appropriate one, leaving the other unchanged.

Reply:
Wait no, because the existing rule (`ruleWithoutAgent`) still applies to agents whose access is unchanged. So either the new agent wasn't part of the original rule, in which case `const ruleWithoutAgent = removeIri(rule, acl.agent, agent);` doesn't impact the existing rule, or the agent whose access is changed was already in the rule, in which case `getIriAll(originalRule, acl.agent).includes(agent);` is true, and the existing access that isn't the subject of the change (e.g. default access for `setAgentResourceAccess`) is left untouched.

Reply:
Ah yes, you're right.

I do think I just thought of another bug though (pre-existing, not introduced by this PR): if a Rule applies to both an Agent and an Agent Group/Class, wouldn't _both_ `ruleWithoutAgent` and `ruleForOtherTargets` apply to those as well? Hence giving whatever access we're giving to the Agent to the Group or Class as well, in addition to the access they've already been given. Which would mean we'd have to call `removeAll(ruleForOtherTargets, acl.agentGroup)` and `removeAll(ruleForOtherTargets, acl.agentClass)` as well to fix it.

Let me know if you think that's correct, then I'll create another ticket/PR to fix it.

Reply:
I added an `expect` to the test in 8126087fe12f4bb4d96e488c102c98b085264de9 to ensure this is true.

Reply:
And yeah that sounds correct. I'll merge the current PR, so that we can merge #295 as well and apply this fix on it directly, since it will likely be impacted